Final Project/Assets/RenderPipeline/RayTracing.compute.compute

```diff
@@ -39,6 +55,48 @@ Ray CreateCameraRay(float2 uv)
     return CreateRay(origin, direction);
 }
 
+void IntersectSphere(Ray ray, inout RayHit bestHit, float4 sphere)
+{
+    // Calculate distance along the ray where the sphere is intersected
+    float3 d = ray.origin - sphere.xyz;
+    float p1 = -dot(ray.direction, d);
+    float p2sqr = p1 * p1 - dot(d, d) + sphere.w * sphere.w;
+    if (p2sqr < 0)
+        return;
+    float p2 = sqrt(p2sqr);
+    float t = p1 - p2 > 0 ? p1 - p2 : p1 + p2;
```

Comment:
Shouldn't this distance calc instead be
`float t = p1 - p2 > 0 ? p1 - p2 : p2 - p1;`

Reply:
Taking a look

Reply:
Because hit distance cannot be negative, and we want the closest hit, so we take the least positive value between p1 and p2. I have updated the code with more detailed explanation